### PR TITLE
Fixing potential problems with resource accounting in BigtableBufferedMutator

### PR DESCRIPTION
**Pull request by sduskis**
Handling a case where the RPC is already complete before it gets registered.

### CODE REVIEW

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java [hunk 1]
```diff
@@ -101,51 +111,121 @@ public long getMaxHeapSize() {
...
-        wait(WAIT_MILLIS);
-      }
-      if (performedWarning) {
-        LOG.info("flush() completed");
+      cleanupCompletedCallbacks();
+      while (!activeRequests.isEmpty()) {
+        waitForCompletions();
       }
+      LOG.trace("flush() completed");
+      nextLogOrSweep = -1L;
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: move this to before while loop?

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java [hunk 2]
```diff
@@ -101,51 +111,121 @@ public long getMaxHeapSize() {
...
+    public synchronized void registerInvokedCallback(RequestAccountant callback) {
+      // There could have been a backlog of RPCs, and the RPC completed by the time the synchronized
+      // registerInvokedCallback method was finally called. In that case, simply clean up the heap
+      // size. Otherwise, keep track of the callback in the activeRequests Map.
+      if (callback.isRequestDone()) {
+        operationComplete(callback);
       } else {
-        LOG.warn("An operation completion was recieved multiple times. Your operations completed."
-            + " Please notify Google that this occurred.");
+        this.activeRequests.put(callback.operationSequenceId, callback);
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: nit: remove "this."

**Comment by AngusDavis** [comment]
I'm a bit confused on the problem. In registerOperationWithHeapSize, if an InterruptedException is thrown then no pending operation will be registered and so the catch block on 350 (now 407), just needs to either throw or exit the method.

If the Channel has given us a Future, even if the future is already complete, calling addCallback will cause the future's callback to be triggered. If that's not happening, then Guava and/or Java's Futures are broken. 

If the Future is never being completed or Failed, then that's a gRPC issue and I'm not certain this will solve that problem.

**Comment by sduskis** [comment]
Closing this in lieu of #322.

**Pull request closed without merging**